src/main/ruby/core/pre.rb

```diff
@@ -57,3 +57,14 @@ def to_sym
     self
   end
 end
+
+if Truffle::Boot.preinitializing?
+  Truffle::Boot::TO_RUN_AT_INIT = []
+  def delay(&block)
+    Truffle::Boot::TO_RUN_AT_INIT << block
+  end
+else
+  def delay
+    yield
+  end
+end
```

Comment:
Surely this needs to be namespaced?

Reply:
The method is removed in `post-boot.rb` so I think it doesn't matter.
But it could be namespaced under `Truffle::Boot` if you think it's clearer when used in the core.

Reply:
Truffle::Boot sounds good

Reply:
I changed it to `Truffle::Boot.delay`.